File: pcode/convert.py
import math
import itertools

def part1(a, base):
    x, y = 0, a
    L = []
    while y:
        L.append(x)
        x, y = y % base, y / base
    L.append(x)
    return L

def part2(b, base):
    x, y = 0, b
    M = []
    while y:
        M.append(x)
        x, y = int(math.floor(y * base)), y * base - int(math.floor(y * base))
    M.append(x)
    return M

def dec_to_any():
    '''This function also only supports non-negative numbers'''

    num = input('Please enter a decimal number:')
    base = input('Please enter the base:')
    L = part1(int(math.floor(num)), base)[1:]
    L.reverse()
    print ''.join(str(i) for i in L) + '.' + ''.join(str(j) for j in part2(num - int(math.floor(num)), base)[1:])

dec_to_any()

File: ccode/convert.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main(void)                                                            // This function only supports non-negative number convertion
{
    double num, final = 0.0, temp2;
    int base, i, temp1;
    int bit[20];
    for(i = 0; i < 20; i++)
        bit[i] = 0;
    printf("Please enter the decimal number:\n");
    scanf("%lf", &num);
    printf("And then the base:\n");
    scanf("%d", &base);
    for(i = 0, temp1 = (int)(floor(num)); ; i++)
    {
         bit[i] = temp1 % base;
         final += bit[i] * pow(10, i);
         temp1 /= base;
         if(temp1 == 0)
             break;
    }
    for(i = 0; i < 20; i++)
        bit[i] = 0;
    for(i = 0, temp2 = num - (int)(floor(num)); i < 20; i++)
    {
        bit[i] = (int)(floor(temp2 * base));
        final += bit[i] * pow(10, -i - 1);
        temp2 = temp2 * base - bit[i];
        if(temp2 == 0)
            break;

    }
    printf("%.8lf\n", final);
    return 0;
}
